tests/lifecycle/rediscache_cases_test.go

```diff
@@ -2,14 +2,197 @@
 
 package lifecycle
 
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/open-service-broker-azure/pkg/generate"
+
+	"github.com/satori/go.uuid"
+
+	networkSDK "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-01-01/network"
+	storageSDK "github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2017-10-01/storage"
+	"github.com/Azure/open-service-broker-azure/pkg/service"
+)
+
 var rediscacheTestCases = []serviceLifecycleTestCase{
 	{
 		group:     "rediscache",
-		name:      "rediscache",
+		name:      "rediscache-basic-provision-and-update",
 		serviceID: "0346088a-d4b2-4478-aa32-f18e295ec1d9",
 		planID:    "362b3d1b-5b57-4289-80ad-4a15a760c29c",
 		provisioningParameters: map[string]interface{}{
-			"location": "southcentralus",
+			"location":         "southcentralus",
+			"skuCapacity":      1,
+			"enableNonSslPort": "disabled",
+		},
+		updatingParameters: map[string]interface{}{
+			"skuCapacity":      2,
+			"enableNonSslPort": "enabled",
 		},
 	},
+	{
+		group:     "rediscache",
+		name:      "rediscache-premium-provision-and-update",
+		serviceID: "0346088a-d4b2-4478-aa32-f18e295ec1d9",
+		planID:    "b1057a8f-9a01-423a-bc35-e168d5c04cf0",
+		provisioningParameters: map[string]interface{}{
+			"location":   "eastus",
+			"shardCount": 2,
+		},
+		updatingParameters: map[string]interface{}{
+			"shardCount":       1,
+			"enableNonSslPort": "disabled",
+			"redisConfiguration": map[string]interface{}{
+				"rdb-backup-enabled": "disabled",
+			},
+		},
+		preProvisionFns: []preProvisionFn{
+			createVirtualNetwork,
+			createStorageAccount,
+		},
+	},
+}
+
+// nolint: lll
+func createVirtualNetwork(
+	ctx context.Context,
+	resourceGroup string,
+	parent *service.Instance,
+	pp *map[string]interface{},
+) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	azureConfig, err := getAzureConfig()
+	if err != nil {
+		return fmt.Errorf("error getting azure config %s", err)
+	}
+	authorizer, err := getBearerTokenAuthorizer(azureConfig)
+	if err != nil {
+		return fmt.Errorf("error getting authorizer %s", err)
+	}
+
+	virtualNetworksClient := networkSDK.NewVirtualNetworksClientWithBaseURI(
+		azureConfig.Environment.ResourceManagerEndpoint,
+		azureConfig.SubscriptionID,
+	)
+	virtualNetworksClient.Authorizer = authorizer
+
+	virtualNetworkName := uuid.NewV4().String()
+	eastus := "eastus"
+	subnetName := "default"
+	subnetCIDR := "172.19.0.0/24"
```

Comment:
Try this to get a string pointer: https://github.com/Azure/go-autorest/blob/master/autorest/to/convert.go#L30.
It is the usage in sdk samples: https://github.com/Azure-Samples/azure-sdk-for-go-samples/blob/master/network/network.go#L38

Reply:
Thanks, will fix it.